gps-nmea/nmea_parser.go

```diff
@@ -0,0 +1,137 @@
+package nmea
+
+import (
+	"math"
+	"strconv"
+	"strings"
+
+	"github.com/adrianmo/go-nmea"
+	geo "github.com/kellydunn/golang-geo"
+	"github.com/pkg/errors"
+	"go.uber.org/multierr"
+)
+
+const (
+	knotsToMPerSec = 0.51444
+	kphToMPerSec   = 0.27778
+)
+
+type gpsData struct {
+	location   *geo.Point
+	alt        float64
+	speed      float64 // ground speed in m per sec
+	vDOP       float64 // vertical accuracy
+	hDOP       float64 // horizontal accuracy
+	satsInView int     // quantity satellites in view
+	satsInUse  int     // quantity satellites in view
+	valid      bool
+	fixQuality int
+}
+
+func errInvalidFix(sentenceType, badFix, goodFix string) error {
+	return errors.Errorf("type %q sentence fix is not valid have: %q  want %q", sentenceType, badFix, goodFix)
+}
+
+// parseAndUpdate will attempt to parse a line to an NMEA sentence, and if valid, will try to update the given struct
+// with the values for that line. Nothing will be updated if there is not a valid gps fix.
+func (g *gpsData) parseAndUpdate(line string) error {
```

Comment:
I think we should eventually export this package and only this package from RDK, so that when we change it to have more capabilities, we're not changing things in two places. 
Next ticket. I think.

Reply:
hmm okay so move the NMEA constructors to the module but still use the nmea_parser in rdk? I think I would need to export this function and gpsData struct in rdk to do that.

Reply:
Yes - and those are okay to export since they are packages that help you do something in runtime rather than things that create and handle resources once the server starts. They don't depend on the structures that are in use for returning API calls, so we'd like them to not be different across modules that do the same thing.

Reply:
okay sounds good! Ill do this and move the nmea constructor to the i2c-rtk package in the next PR.